Integrations/integration-pyEWS.yml

```diff
@@ -0,0 +1,1137 @@
+
+        def get_payload(self, query, mailboxes):
+            def get_mailbox_search_scope(mailbox_id):
+                mailbox_search_scope = create_element("t:MailboxSearchScope")
+                add_xml_child(mailbox_search_scope, "t:Mailbox", mailbox_id)
+                add_xml_child(mailbox_search_scope, "t:SearchScope", "All")
+                return mailbox_search_scope
+
+            mailbox_query_element = create_element("t:MailboxQuery")
+            add_xml_child(mailbox_query_element, "t:Query", query)
+            mailboxes_scopes = []
+            for mailbox in mailboxes:
+                mailboxes_scopes.append(get_mailbox_search_scope(mailbox))
+            add_xml_child(mailbox_query_element, "t:MailboxSearchScopes", mailboxes_scopes)
+
+            element = create_element('m:%s' % self.SERVICE_NAME)
+            add_xml_child(element, "m:SearchQueries", mailbox_query_element)
+            add_xml_child(element, "m:ResultType", "PreviewOnly")
+
+            return element
+
+
+    def get_searchable_mailboxes(protocol):
+        searchable_mailboxes = GetSearchableMailboxes(protocol=protocol).call()
+        return get_entry_for_object("Searchable mailboxes", 'EWS.Mailboxes', searchable_mailboxes)
+
+
+    def search_mailboxes(protocol, filter, mailbox_search_scope=None):
+        if not mailbox_search_scope:
+            entry = get_searchable_mailboxes(protocol)
+            mailboxes = entry[ENTRY_CONTEXT]['EWS.Mailboxes']
+            mailbox_search_scope = map(lambda x: x[MAILBOX_ID], mailboxes)
+        search_results = SearchMailboxes(protocol=protocol).call(filter, mailbox_search_scope)
+        return get_entry_for_object("Search mailboxes results",
+                                    CONTEXT_UPDATE_EWS_ITEM,
+                                    search_results)
+
+
+    def get_account(account_email, access_type=ACCESS_TYPE):
+        return Account(
+            primary_smtp_address=account_email, autodiscover=False, config=config, access_type=access_type,
+        )
+
+
+    def get_last_run():
+        last_run = demisto.getLastRun()
+        if not last_run or last_run.get(LAST_RUN_FOLDER) != FOLDER_NAME:
+            last_run = {
+                LAST_RUN_TIME: None,
+                LAST_RUN_FOLDER: FOLDER_NAME,
+                LAST_RUN_IDS: None
+            }
+        if LAST_RUN_TIME in last_run and last_run[LAST_RUN_TIME] is not None:
+            last_run[LAST_RUN_TIME] = EWSDateTime.from_string(last_run[LAST_RUN_TIME])
+
+        return last_run
+
+
+    def fetch_last_emails(account, folder_name='Inbox', since_datetime=None, exclude_ids=None):
+        qs = get_folder_by_path(account.root, folder_name)
+        if since_datetime:
+            qs = qs.filter(datetime_received__gte=since_datetime)
+        else:
+            if not FETCH_ALL_HISTORY:
+                last_10_min = EWSDateTime.now(tz=EWSTimeZone.timezone('UTC')) - timedelta(minutes=10)
```

Comment:
what EWS is in a different timezone from Demisto server?
Do you support such configuration here?

Reply:
demisto timezone does not matter. EWSDateTime is always with timezone. the lib take care of it.